cpg_workflows/jobs/count.py

```diff
@@ -0,0 +1,222 @@
+    """
+    Count RNA seq reads mapping to genes and/or transcripts using featureCounts.
+    """
+    # Reuse existing output if possible
+    if (
+        output_path and
+        summary_path and
+        can_reuse(output_path, overwrite) and
+        can_reuse(summary_path, overwrite)
+    ):
+        return []
+    
+    jobs: list[Job] = []
+
+    # Determine whether input is a BAM file or a CRAM file
+    if isinstance(input_cram_or_bam, BamPath):
+        # Localise input
+        input_bam_reads = b.read_input_group(**{
+            'bam': str(input_cram_or_bam.path),
+            'bam.bai': str(input_cram_or_bam.index_path),
+        })
+    elif isinstance(input_cram_or_bam, CramPath):
+        # Localise input
+        input_cram_reads = b.read_input_group(**{
+            'cram': str(input_cram_or_bam.path),
+            'cram.crai': str(input_cram_or_bam.index_path),
+        })
+        # Convert CRAM to BAM
+        j, input_bam_reads = cram_to_bam(
+            b=b,
+            input_cram=input_cram_reads,
+            output_bam=cram_to_bam_path,
+            job_attrs=job_attrs,
+            requested_nthreads=requested_nthreads,
+        )
+        if j and isinstance(j, Job):
+            jobs.append(j)
+    else:
+        raise ValueError(
+            f'Invalid alignment input: "{str(input_cram_or_bam)}", expected BAM or CRAM file.'
+        )
+    
+    assert isinstance(input_bam_reads, hb.ResourceGroup)
+
+    counting_reference = count_res_group(b)
+
+    # Create job
+    job_name = f'count_{sample_name}' if sample_name else 'count'
+    _job_attrs = (job_attrs or {}) | dict(label=job_name, tool='featureCounts')
+    j = b.new_job(job_name, _job_attrs)
+    j.image(image_path('subread'))
+    
+    # Set resource requirements
+    nthreads = requested_nthreads or 8
+    res = STANDARD.set_resources(
+        j,
+        ncpu=nthreads,
+        storage_gb=50,  # TODO: make configurable
+    )
+
+    # Declare output resource group
+    j.declare_resource_group(
+        count_output={
+            'count': '{root}.count',
+            'count.summary': '{root}.count.summary',
+        }
+    )
+    
+    # Create counting command
+    fc = FeatureCounts(
+        input_bam=input_bam_reads.bam,
+        gtf_file=counting_reference.gtf,
+        output_path=j.count_output['count'],
+        summary_path=j.count_output['count.summary'],
+        paired_end=True,
+        feature_type='exon',
+        attribute='gene_id',
+        strandness='reverse',  # TODO: confirm this
+        multi_mapping=False,  # TODO: determine default value
+        min_quality=None,  # TODO: determine default value
+        primary_only=True,  # TODO: determine default value
+        ignore_duplicates=False,  # TODO: determine default value
+        count_pairs=True,  # TODO: determine default value
+        both_ends_mapped=True,  # TODO: determine default value
+        both_ends_same_chr=True,  # TODO: determine default value
```

Comment:
Are these TODOs all to do, or are they settled?

Reply:
Ah, thanks for pointing that out, I think they're good defaults, but I will also plan to update the code in a future edit to expose these settings through the config toml.